usl/usl_kevm.h

```diff
@@ -0,0 +1,241 @@
+};
+using access_list_t = std::vector<access_pair_t>;
+
+// Transaction Type
+enum class tx_type_t { LEGACY, ACCESS_LIST, DYNAMIC_FEE };
+using tx_type_option_t = std::optional<tx_type_t>;
+
+// Message
+using account_address_option_t = std::optional<account_address_t>;
+using sig_v_t = std::uint64_t;
+struct message_t {
+  message_id_t message_id;
+  nonce_t tx_nonce;
+  ether_t tx_gas_price;
+  gas_t tx_gas_limit;
+  account_address_option_t to;
+  ether_t value;
+  sig_v_t sig_v;
+  bytes_t sig_r;
+  bytes_t sig_s;
+  bytes_t data;
+  access_list_t tx_access;
+  chain_id_t tx_chain_id;
+  ether_t tx_priority_fee;
+  ether_t tx_max_fee;
+  tx_type_option_t tx_type;
+};
+
+// Network Callbacks
+using add_account_callback_t = void (*)(account_t);
+using remove_account_callback_t = void (*)(account_address_t);
+
+using get_account_balance_callback_t =
+    std::optional<ether_t> (*)(account_address_t);
+using set_account_balance_callback_t = void (*)(account_address_t, ether_t);
+
+using get_account_code_callback_t =
+    std::optional<bytes_t> (*)(account_address_t);
+using set_account_code_callback_t = void (*)(account_address_t, bytes_t);
+
+using get_account_nonce_callback_t =
+    std::optional<nonce_t> (*)(account_address_t);
+using set_account_nonce_callback_t = void (*)(account_address_t, nonce_t);
+
+using get_account_storage_callback_t =
+    std::optional<storage_value_t> (*)(account_address_t, storage_key_t);
+using set_account_storage_callback_t = void (*)(account_address_t,
+                                                storage_key_t, storage_value_t);
+
+using get_account_orig_storage_callback_t =
+    std::optional<storage_value_t> (*)(account_address_t, storage_key_t);
+using set_account_orig_storage_callback_t = void (*)(account_address_t,
+                                                     storage_key_t,
+                                                     storage_value_t);
+
+using add_message_callback_t = void (*)(message_t);
+using get_message_callback_t = std::optional<message_t> (*)(message_id_t);
+
+using get_blockhash_callback_t = hash_t (*)(size_t offset);
+
+// Network
+struct network_t {
+  chain_id_t chain_id;
+  add_account_callback_t add_account;
+  remove_account_callback_t remove_account;
+  get_account_balance_callback_t get_account_balance;
+  set_account_balance_callback_t set_account_balance;
+  get_account_code_callback_t get_account_code;
+  set_account_code_callback_t set_account_code;
+  get_account_nonce_callback_t get_account_nonce;
+  set_account_nonce_callback_t set_account_nonce;
+  get_account_storage_callback_t get_account_storage;
+  set_account_storage_callback_t set_account_storage;
+  get_account_orig_storage_callback_t get_account_orig_storage;
+  set_account_orig_storage_callback_t set_account_orig_storage;
+  add_message_callback_t add_message;
+  get_message_callback_t get_message;
+  get_blockhash_callback_t get_blockhash;
+};
+
+// Log
+using log_topic_t = bytes_t;
```

Comment:
I believe log topics are u256 or char[32] as well but I would recommend double checking.

Reply:
I checked and you are right. Added changed the type to the placeholder `uint64_t` and added comment.